### PR TITLE
FEATURE: allow admin to delete all posts by a user irrespectively

### PR DESCRIPTION
**Pull request by techAPJ**
This commit allows admin to delete all posts by a user irrespective of site settings `delete_user_max_post_age` and `delete_all_posts_max`.

Screen Shot 2021-08-24 at 17 15 01

### CODE REVIEW

File: app/assets/javascripts/admin/addon/controllers/admin-user-index.js [hunk 1]
```diff
@@ -626,5 +559,57 @@ export default Controller.extend(CanCheckEmails, {
...
+                deletedPercentage: deletedPercentage,
+              });
+              performDelete(progressModal);
+            }
+          })
+          .catch((e) => {
+            progressModal.send("closeModal");
+            let error;
+            AdminUser.find(user.get("id")).then((u) => user.setProperties(u));
+            if (e.jqXHR.responseJSON && e.jqXHR.responseJSON.errors) {
```
Comment (eviltrout) [inline app/assets/javascripts/admin/addon/controllers/admin-user-index.js:597]: Can we not use `extractError` on this with a `defaultMessage`?  From `lib/ajax-error`

File: app/assets/javascripts/admin/addon/controllers/admin-user-index.js [hunk 2]
```diff
@@ -626,5 +559,57 @@ export default Controller.extend(CanCheckEmails, {
...
+              error = e.jqXHR.responseJSON.errors[0];
+            }
+            error = error || I18n.t("admin.user.delete_posts_failed");
+            bootbox.alert(error);
+          });
+      };
+
+      const openProgressModal = () => {
```
Comment (eviltrout) [inline app/assets/javascripts/admin/addon/controllers/admin-user-index.js:605]: I don't understand why you are making a closure here and then calling it immediately afterwards. Could you not set `const progressModal = showModal("admin-delete-user-posts-progress", { admin: true })` as one line?

**Review by eviltrout — APPROVED** [review]
Looks good. Just a couple minor comments.

File: app/assets/javascripts/admin/addon/controllers/admin-user-index.js (hunk 1 above)
Comment (techAPJ) [reply app/assets/javascripts/admin/addon/controllers/admin-user-index.js]: Sure, I've made this change. Thanks for the tip!

File: app/assets/javascripts/admin/addon/controllers/admin-user-index.js (hunk 2 above)
Comment (techAPJ) [reply app/assets/javascripts/admin/addon/controllers/admin-user-index.js]: I am not sure why I did that, I was trying to reuse the original code but forgot to optimize it. I've now made the change you suggested.

File: lib/guardian/post_guardian.rb [hunk 3]
```diff
@@ -100,8 +100,9 @@ def can_delete_all_posts?(user)
     is_staff? &&
     user &&
     !user.admin? &&
-    (user.first_post_created_at.nil? || user.first_post_created_at >= SiteSetting.delete_user_max_post_age.days.ago) &&
-    user.post_count <= SiteSetting.delete_all_posts_max.to_i
+    (is_admin? ||
```
Comment (tgxworld) [inline lib/guardian/post_guardian.rb:103]: The `is_admin?` condition doesn't look right to me. There is a `!user.admin?` check before so `is_admin?` will never be `true`

**Pull request merged**